### PR TITLE
[fix](auto-partition) fix auto partition load lost data in multi sender

### PR DESCRIPTION
**Pull request by kaijchen**
## Proposed changes

Change `use_cnt` mechanism for incremental (auto partition) channels and streams, it's now dynamically counted.
Use `close_wait()` of regular partitions as a synchronize point to make sure all sinks are in close phase before closing any incremental (auto partition) channels and streams.
Add dummy (fake) partition and tablet if there is no regular partition in the auto partition table.

Replace #34740

Co-authored-by: zhaochangle <zhaochangle@selectdb.com>

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

**Comment by kaijchen** [comment]
run buildall

File: be/src/vec/sink/writer/vtablet_writer_v2.cpp [hunk 1]
```diff
@@ -311,6 +311,10 @@ Status VTabletWriterV2::_build_tablet_node_mapping() {
                     tablet.set_index_id(index.index_id);
                     tablet.set_tablet_id(tablet_id);
                     _tablets_for_node[node].emplace(tablet_id, tablet);
+                    if (tablet_id == 0) [[unlikely]] {
```
Comment (kaijchen) [inline be/src/vec/sink/writer/vtablet_writer_v2.cpp:314]:
```suggestion
                    constexpr int64_t DUMMY_TABLET_ID = 0;
                    if (tablet_id == DUMMY_TABLET_ID) [[unlikely]] {
```

**Comment by kaijchen** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

File: regression-test/suites/partition_p1/auto_partition/sql/multi_thread_load.groovy [hunk 2]
```diff
@@ -19,7 +19,7 @@ import groovy.io.FileType
 import java.nio.file.Files
 import java.nio.file.Paths
 
-suite("multi_thread_load", "p1,nonConcurrent") { // stress case should use resource fully
+suite("multi_thread_load") {
```
Comment (zclllyybb) [inline regression-test/suites/partition_p1/auto_partition/sql/multi_thread_load.groovy:22]: please revert this line change

Comment (kaijchen) [inline regression-test/suites/partition_p1/auto_partition/sql/multi_thread_load.groovy:22]:
```suggestion
suite("multi_thread_load", "p1,nonConcurrent") { // stress case should use resource fully```

**Comment by kaijchen** [comment]
run buildall

**Review by zclllyybb — APPROVED** [review]
LGTM

**Review by liaoxin01 — APPROVED** [review]
LGTM

**Pull request merged**